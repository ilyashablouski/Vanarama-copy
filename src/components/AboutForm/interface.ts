import { AboutFormDropdownData } from '../../../generated/AboutFormDropdownData';
import { AboutFormPerson } from '../../../generated/AboutFormPerson';
import { IExistenceCheckResult } from '../AboutForm/mapEmailErrorMessage';

export interface IAboutFormValues {
  title: string;
  firstName: string;
  lastName: string;
  email: string;
  telephone: string;
  dayOfBirth: string;
  monthOfBirth: string;
  yearOfBirth: string;
  countryOfBirth: string;
  nationality: string;
  maritalStatus: string;
  dependants: string;
  adultsInHousehold: string;
  termsAndCons: boolean;
  consent?: boolean;
  termsAndConditions?: boolean;
  marketing?: boolean;
  companyType?: string;
  privacyPolicy?: boolean;
}

export interface IProps {
  dropdownData: AboutFormDropdownData;
  person?: AboutFormPerson | null;
  submit: (values: IAboutFormValues) => Promise<any>;
  onEmailExistenceCheck?: (
    value: string,
  ) => Promise<IExistenceCheckResult | null>;
  onLogInClick?: () => void;
  onRegistrationClick?: () => void;
  personLoggedIn?: boolean;
}
