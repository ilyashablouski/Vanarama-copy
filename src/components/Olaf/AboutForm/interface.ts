export interface IDetails {
    title: string;
    firstName: string;
    lastName: string;
    email: string;
    mobile: string;
    dayOfBirth: string;
    monthOfBirth: string;
    yearOfBirth: string;
    countryOfBirth: string;
    nationality: string;
    maritalStatus: string;
    dependants: string;
    adultsInHousehold: string;
    termsAndCons: boolean;
    consent: boolean;
  }
  
  export interface IState {
    details: IDetails;
    allDropDowns: any;
  }

  export interface IProps {
    captchaFormData: (pageRef: string, data: {}) => void;
  }