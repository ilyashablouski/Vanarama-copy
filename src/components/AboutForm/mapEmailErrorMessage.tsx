import Text from '@vanarama/uibook/lib/components/atoms/text';
import RouterLink from '../RouterLink/RouterLink';

export const EMAIL_ALREADY_EXISTS = 'EMAIL_ALREADY_EXISTS';

const logInLink = {
  href: '',
  label: 'login',
};

export const mapEmailErrorMessage = (onClick?: () => void, message?: string) =>
  message !== EMAIL_ALREADY_EXISTS ? (
    message
  ) : (
    <Text tag="p" color="danger" size="xsmall">
      {'Your email address already exists. Do you wish to '}
      <RouterLink
        link={logInLink}
        onClick={onClick}
        classNames={{
          color: 'teal',
          size: 'xsmall',
        }}
      />
      {' using this email?'}
    </Text>
  );
