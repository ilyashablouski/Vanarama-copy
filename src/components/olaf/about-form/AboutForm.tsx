import { FC } from 'react';
import { Row, Col } from 'react-grid-system';

import Heading from '@vanarama/uibook/packages/ui-components/src/components/atoms/heading';
import Text from '@vanarama/uibook/packages/ui-components/src/components/atoms/text';
import FormGroup from '@vanarama/uibook/packages/ui-components/src/components/molecules/formgroup';
import Select from '@vanarama/uibook/packages/ui-components/src/components/atoms/select/';
import TextInput from '@vanarama/uibook/packages/ui-components/src/components/atoms/textinput/';
import Button from '@vanarama/uibook/packages/ui-components/src/components/atoms/button/';
import ChevronForwardSharpIcon from '@vanarama/uibook/packages/ui-components/src/assets/icons/ChevronForwardCircleSharp';
import CheckBox from '@vanarama/uibook/packages/ui-components/src/components/atoms/checkbox/';
import { Controller, useForm } from 'react-hook-form';
import { genMonths, genYears } from '../../../services/utils/helpers';
import { IProps, IDetails } from './interface';
import validationSchema from './AboutForm.validation';

const AboutForm: FC<IProps> = ({ allDropDowns = {}, submit }) => {
  const months: string[] = genMonths() || [];
  const years: number[] = genYears(100) || [];

  const { handleSubmit, errors, control, reset, register } = useForm<IDetails>({
    mode: 'onBlur',
    validationSchema,
  });

  const onSubmission = (values: IDetails) => {
    submit(values);
    reset();
  };

  return (
    <form onSubmit={handleSubmit(onSubmission)} id="aboutForm" className="form">
      <Heading color="black" size="xlarge">
        About You
      </Heading>
      <Text color="darker" size="lead">
        We just need some initial details for your credit check.
      </Text>
      <FormGroup legend="Title">
        <Controller
          name="title"
          options={allDropDowns.titles}
          data-testid="aboutTitle"
          as={Select}
          control={control}
        />
      </FormGroup>
      <FormGroup>
        <Row>
          <Col sm={8}>
            <TextInput
              name="firstName"
              label="First Name"
              type="text"
              data-testid="aboutFirstName"
              parentRef={register}
              invalid={errors?.firstName?.message}
            />
          </Col>
        </Row>
      </FormGroup>
      <FormGroup>
        <Row>
          <Col sm={8}>
            <TextInput
              label="Last Name"
              type="text"
              name="lastName"
              data-testid="aboutLastName"
              parentRef={register}
              invalid={errors?.lastName?.message}
            />
          </Col>
        </Row>
      </FormGroup>
      <FormGroup>
        <Row>
          <Col sm={6}>
            <TextInput
              label="Email"
              type="email"
              name="email"
              data-testid="aboutEmail"
              parentRef={register}
              invalid={errors?.email?.message}
            />
          </Col>
        </Row>
      </FormGroup>
      <FormGroup>
        <Row>
          <Col sm={6}>
            <TextInput
              label="Mobile"
              type="tel"
              name="mobile"
              data-testid="aboutMobile"
              parentRef={register}
              invalid={errors?.mobile?.message}
            />
          </Col>
        </Row>
      </FormGroup>
      <FormGroup legend="Date of Birth">
        <Row>
          <Col sm={4}>
            <Controller
              data-testid="aboutSelectDOB"
              name="dayOfBirth"
              as={Select}
              control={control}
              options={{
                data: [...Array(31)].map((_, i) => (i + 1).toString()),
              }}
            />
          </Col>
          <Col sm={4}>
            <Controller
              data-testid="aboutSelectMOB"
              name="monthOfBirth"
              as={Select}
              control={control}
              options={{
                data: months,
              }}
            />
          </Col>
          <Col sm={4}>
            <Controller
              data-testid="aboutSelectYOB"
              name="yearOfBirth"
              as={Select}
              control={control}
              options={{
                data: years.map(year => year.toString()),
              }}
            />
          </Col>
        </Row>
      </FormGroup>
      <FormGroup legend="Country of Birth">
        <Controller
          name="countryOfBirth"
          as={Select}
          control={control}
          options={allDropDowns.countries}
          data-testid="aboutSelectCOB"
        />
      </FormGroup>
      <FormGroup legend="Nationality">
        <Controller
          name="nationality"
          as={Select}
          control={control}
          options={allDropDowns.nationalities}
          data-testid="aboutNationality"
        />
      </FormGroup>
      <FormGroup legend="Marital Status">
        <Controller
          name="maritalStatus"
          as={Select}
          control={control}
          options={allDropDowns.maritalStatuses}
          data-testid="aboutMaritalStatus"
        />
      </FormGroup>
      <FormGroup legend="No. of Dependants">
        <Controller
          name="dependants"
          as={Select}
          control={control}
          options={allDropDowns.noOfDependants}
          data-testid="aboutDependants"
        />
      </FormGroup>
      <FormGroup legend="No. of Adults in Household">
        <Controller
          name="adultsInHousehold"
          as={Select}
          control={control}
          options={allDropDowns.noOfAdultsInHousehold}
          data-testid="aboutAdultsInHouse"
        />
      </FormGroup>
      <FormGroup legend="Please Confirm">
        <Controller
          id="consent"
          data-testid="aboutConsent"
          as={CheckBox}
          control={control}
          name="consent"
          label="I wish to receive emails and SMS messages for updates on the latest deals, offers and promotions."
          defaultValue={false}
        />
        <Controller
          id="termsAndCons"
          data-testid="aboutTermsAndCons"
          as={CheckBox}
          control={control}
          name="termsAndCons"
          label="agree to the terms and conditions."
          defaultValue={false}
        />
      </FormGroup>

      <Button
        type="submit"
        label="Continue"
        color="primary"
        icon={<ChevronForwardSharpIcon />}
        iconPosition="after"
      />
    </form>
  );
};

export default AboutForm;
