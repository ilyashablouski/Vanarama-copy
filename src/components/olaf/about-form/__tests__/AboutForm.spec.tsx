import React from 'react';
import {
  fireEvent,
  render,
  waitFor,
  screen,
  cleanup,
} from '@testing-library/react';
import AboutForm from '..';

describe.skip('<AboutForm />', () => {
  const submit = jest.fn();
  let rendered;

  beforeEach(() => {
    const { container } = render(
      <AboutForm submit={submit} allDropDowns={{}} preloadData={{}} />,
    );
    rendered = container;
  });

  afterEach(cleanup);

  it('renders with default props correctly', () => {
    expect(rendered).toMatchSnapshot();
  });

  it('should show required form field validation messages', async () => {
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(screen.getByText('Please select your a title')).toBeVisible();
      expect(screen.getByText('Please enter your first name')).toBeVisible();
      expect(screen.getByText('Please enter your last name')).toBeVisible();
      expect(screen.getByText('Pleas enter your email address')).toBeVisible();
      expect(screen.getByText('Please enter your mobile')).toBeVisible();
      expect(
        screen.getByText('Please complete your date of birth'),
      ).toBeVisible();
      expect(
        screen.getByText('Please enter your country of birth'),
      ).toBeVisible();
      expect(screen.getByText('Please enter your nationality')).toBeVisible();
      expect(
        screen.getByText('Please enter your marital status'),
      ).toBeVisible();
      expect(
        screen.getByText('Please enter number of dependants'),
      ).toBeVisible();
      expect(
        screen.getByText('Please enter number of adults in household'),
      ).toBeVisible();
      expect(
        screen.getByText('Please confirm terms and conditions'),
      ).toBeVisible();
    });
  });

  it('should assure minimum characters allowed for first name', async () => {
    fireEvent.change(screen.getByTestId('aboutFirstName'), {
      target: { value: 'W' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(
          'Oops, this name’s too short. Please make it longer than 2 characters',
        ),
      );
    });
  });

  it('should assure max characters allowed for first name', async () => {
    fireEvent.change(screen.getByTestId('aboutFirstName'), {
      target: { value: 'ThisFirstNameIsOverFiftyCharactersLongggggggggggggg' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(
          'Oops, this name’s too long. Please keep it to 50 characters',
        ),
      );
    });
  });

  it('should assure minimum characters allowed for last name', async () => {
    fireEvent.change(screen.getByTestId('aboutLastName'), {
      target: { value: 'W' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(
          'Oops, this name’s too short. Please make it longer than 2 characters',
        ),
      );
    });
  });

  it('should assure max characters allowed for last name', async () => {
    fireEvent.change(screen.getByTestId('aboutLastName'), {
      target: { value: 'ThisLastNameIsOverFiftyCharactersLonggggggggggggggg' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(
          'Oops, this name’s too long. Please keep it to 50 characters',
        ),
      );
    });
  });

  it('should assure correct email format', async () => {
    fireEvent.change(screen.getByTestId('aboutEmail'), {
      target: { value: 'invalid@' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(screen.getByText('Oops, this email address is invalid'));
    });
  });

  it('should assure min mobile chars', async () => {
    fireEvent.change(screen.getByTestId('aboutMobile'), {
      target: { value: '0740292992' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(
          'Please enter mobile number without spaces or hyphens',
        ),
      );
    });
  });

  it('should assure max mobile chars', async () => {
    fireEvent.change(screen.getByTestId('aboutMobile'), {
      // over 15 digits
      target: { value: '0740292992222222' },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(
        screen.getByText(
          'Please enter mobile number without spaces or hyphens',
        ),
      );
    });
  });

  it('should assure minimum age', async () => {
    const thisYear = new Date().getFullYear();
    fireEvent.change(screen.getByTestId('aboutSelectDOB'), {
      target: { value: '1' },
    });
    fireEvent.change(screen.getByTestId('aboutSelectMOB'), {
      target: { value: 'January' },
    });
    fireEvent.change(screen.getByTestId('aboutSelectYOB'), {
      target: { value: thisYear },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => {
      expect(screen.getByText('Oops, you’re too young.'));
    });
  });

  it('should call submit with valid field inputs', async () => {
    fireEvent.change(screen.getByTestId('aboutFirstName'), {
      target: { value: 'John' },
    });
    fireEvent.change(screen.getByTestId('aboutLastName'), {
      target: { value: 'Doe' },
    });
    fireEvent.change(screen.getByTestId('aboutEmail'), {
      target: { value: 'hello@email.com' },
    });
    fireEvent.change(screen.getByTestId('aboutMobile'), {
      target: { value: '07402020200' },
    });
    fireEvent.change(screen.getByTestId('aboutSelectDOB'), {
      target: { value: '1' },
    });
    fireEvent.change(screen.getByTestId('aboutSelectMOB'), {
      target: { value: 'January' },
    });
    fireEvent.change(screen.getByTestId('aboutSelectYOB'), {
      target: { value: '2000' },
    });
    fireEvent.change(screen.getByTestId('countryOfBirth'), {
      target: { value: 'United Kingdom' },
    });
    fireEvent.change(screen.getByTestId('aboutNationality'), {
      target: { value: 'British' },
    });
    fireEvent.change(screen.getByTestId('aboutMaritalStatus'), {
      target: { value: 'British' },
    });
    fireEvent.change(screen.getByTestId('aboutDependants'), {
      target: { value: '1' },
    });
    fireEvent.change(screen.getByTestId('aboutAdultsInHouse'), {
      target: { value: '1' },
    });
    fireEvent.change(screen.getByTestId('aboutTermsAndCons'), {
      target: { checked: true },
    });
    fireEvent.click(screen.getByText('Continue'));

    // ASSERT
    await waitFor(() => expect(submit).toHaveBeenCalledTimes(1));
  });
});
