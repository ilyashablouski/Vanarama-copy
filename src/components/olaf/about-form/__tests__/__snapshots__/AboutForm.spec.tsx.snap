// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutForm /> renders with default props correctly 1`] = `
<div>
  <form
    class="form"
    id="aboutForm"
  >
    <p
      class="heading -black -xlarge"
    >
      About You
    </p>
    <span
      class="text -lead -darker"
    >
      We just need some initial details for your credit check.
    </span>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        Title
      </legend>
      <div
        class="formgroup--content"
      >
        <select
          class="select--native"
          disabled=""
          name="title"
        >
          <option>
            No options
          </option>
        </select>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <div
        class="formgroup--content"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 66.66666666666667%; flex-basis: 66.66666666666667%; flex-grow: 0; flex-shrink: 0; max-width: 66.66666666666667%; margin-left: 0%;"
          >
            <div
              class="textinput--wrapper"
            >
              <label
                class="text textinput--label -regular -darker"
              >
                First Name
              </label>
              <div
                class="textinput--field-wrapper"
              >
                <input
                  class="textinput"
                  name="firstName"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <div
        class="formgroup--content"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 66.66666666666667%; flex-basis: 66.66666666666667%; flex-grow: 0; flex-shrink: 0; max-width: 66.66666666666667%; margin-left: 0%;"
          >
            <div
              class="textinput--wrapper"
            >
              <label
                class="text textinput--label -regular -darker"
              >
                Last Name
              </label>
              <div
                class="textinput--field-wrapper"
              >
                <input
                  class="textinput"
                  name="lastName"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <div
        class="formgroup--content"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
          >
            <div
              class="textinput--wrapper"
            >
              <label
                class="text textinput--label -regular -darker"
              >
                Email
              </label>
              <div
                class="textinput--field-wrapper"
              >
                <input
                  class="textinput"
                  name="email"
                  type="email"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <div
        class="formgroup--content"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 50%; flex-basis: 50%; flex-grow: 0; flex-shrink: 0; max-width: 50%; margin-left: 0%;"
          >
            <div
              class="textinput--wrapper"
            >
              <label
                class="text textinput--label -regular -darker"
              >
                Mobile
              </label>
              <div
                class="textinput--field-wrapper"
              >
                <input
                  class="textinput"
                  name="mobile"
                  type="tel"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        Date of Birth
      </legend>
      <div
        class="formgroup--content"
      >
        <div
          style="margin-left: -15px; margin-right: -15px; display: flex; flex-wrap: wrap; flex-grow: 0; flex-shrink: 0; align-items: normal; justify-content: flex-start;"
        >
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 33.333333333333336%; flex-basis: 33.333333333333336%; flex-grow: 0; flex-shrink: 0; max-width: 33.333333333333336%; margin-left: 0%;"
          >
            <div
              class="select"
            >
              <select
                class="select--native"
                name="dayOfBirth"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="1"
                >
                  1
                </option>
                <option
                  value="2"
                >
                  2
                </option>
                <option
                  value="3"
                >
                  3
                </option>
                <option
                  value="4"
                >
                  4
                </option>
                <option
                  value="5"
                >
                  5
                </option>
                <option
                  value="6"
                >
                  6
                </option>
                <option
                  value="7"
                >
                  7
                </option>
                <option
                  value="8"
                >
                  8
                </option>
                <option
                  value="9"
                >
                  9
                </option>
                <option
                  value="10"
                >
                  10
                </option>
                <option
                  value="11"
                >
                  11
                </option>
                <option
                  value="12"
                >
                  12
                </option>
                <option
                  value="13"
                >
                  13
                </option>
                <option
                  value="14"
                >
                  14
                </option>
                <option
                  value="15"
                >
                  15
                </option>
                <option
                  value="16"
                >
                  16
                </option>
                <option
                  value="17"
                >
                  17
                </option>
                <option
                  value="18"
                >
                  18
                </option>
                <option
                  value="19"
                >
                  19
                </option>
                <option
                  value="20"
                >
                  20
                </option>
                <option
                  value="21"
                >
                  21
                </option>
                <option
                  value="22"
                >
                  22
                </option>
                <option
                  value="23"
                >
                  23
                </option>
                <option
                  value="24"
                >
                  24
                </option>
                <option
                  value="25"
                >
                  25
                </option>
                <option
                  value="26"
                >
                  26
                </option>
                <option
                  value="27"
                >
                  27
                </option>
                <option
                  value="28"
                >
                  28
                </option>
                <option
                  value="29"
                >
                  29
                </option>
                <option
                  value="30"
                >
                  30
                </option>
                <option
                  value="31"
                >
                  31
                </option>
              </select>
              <span
                class="icon select--chevron"
              >
                <i
                  class="icon -stroke -dark"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <title>
                      ionicons-v5-a
                    </title>
                    <polyline
                      points="112 184 256 328 400 184"
                      style="fill: none; stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 33.333333333333336%; flex-basis: 33.333333333333336%; flex-grow: 0; flex-shrink: 0; max-width: 33.333333333333336%; margin-left: 0%;"
          >
            <div
              class="select"
            >
              <select
                class="select--native"
                name="monthOfBirth"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="January"
                >
                  January
                </option>
                <option
                  value="February"
                >
                  February
                </option>
                <option
                  value="March"
                >
                  March
                </option>
                <option
                  value="April"
                >
                  April
                </option>
                <option
                  value="May"
                >
                  May
                </option>
                <option
                  value="June"
                >
                  June
                </option>
                <option
                  value="July"
                >
                  July
                </option>
                <option
                  value="August"
                >
                  August
                </option>
                <option
                  value="September"
                >
                  September
                </option>
                <option
                  value="October"
                >
                  October
                </option>
                <option
                  value="November"
                >
                  November
                </option>
                <option
                  value="December"
                >
                  December
                </option>
              </select>
              <span
                class="icon select--chevron"
              >
                <i
                  class="icon -stroke -dark"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <title>
                      ionicons-v5-a
                    </title>
                    <polyline
                      points="112 184 256 328 400 184"
                      style="fill: none; stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
          <div
            style="box-sizing: border-box; min-height: 1px; position: relative; padding-left: 15px; padding-right: 15px; width: 33.333333333333336%; flex-basis: 33.333333333333336%; flex-grow: 0; flex-shrink: 0; max-width: 33.333333333333336%; margin-left: 0%;"
          >
            <div
              class="select"
            >
              <select
                class="select--native"
                name="yearOfBirth"
              >
                <option
                  value=""
                >
                  Please Select
                </option>
                <option
                  value="1921"
                >
                  1921
                </option>
                <option
                  value="1922"
                >
                  1922
                </option>
                <option
                  value="1923"
                >
                  1923
                </option>
                <option
                  value="1924"
                >
                  1924
                </option>
                <option
                  value="1925"
                >
                  1925
                </option>
                <option
                  value="1926"
                >
                  1926
                </option>
                <option
                  value="1927"
                >
                  1927
                </option>
                <option
                  value="1928"
                >
                  1928
                </option>
                <option
                  value="1929"
                >
                  1929
                </option>
                <option
                  value="1930"
                >
                  1930
                </option>
                <option
                  value="1931"
                >
                  1931
                </option>
                <option
                  value="1932"
                >
                  1932
                </option>
                <option
                  value="1933"
                >
                  1933
                </option>
                <option
                  value="1934"
                >
                  1934
                </option>
                <option
                  value="1935"
                >
                  1935
                </option>
                <option
                  value="1936"
                >
                  1936
                </option>
                <option
                  value="1937"
                >
                  1937
                </option>
                <option
                  value="1938"
                >
                  1938
                </option>
                <option
                  value="1939"
                >
                  1939
                </option>
                <option
                  value="1940"
                >
                  1940
                </option>
                <option
                  value="1941"
                >
                  1941
                </option>
                <option
                  value="1942"
                >
                  1942
                </option>
                <option
                  value="1943"
                >
                  1943
                </option>
                <option
                  value="1944"
                >
                  1944
                </option>
                <option
                  value="1945"
                >
                  1945
                </option>
                <option
                  value="1946"
                >
                  1946
                </option>
                <option
                  value="1947"
                >
                  1947
                </option>
                <option
                  value="1948"
                >
                  1948
                </option>
                <option
                  value="1949"
                >
                  1949
                </option>
                <option
                  value="1950"
                >
                  1950
                </option>
                <option
                  value="1951"
                >
                  1951
                </option>
                <option
                  value="1952"
                >
                  1952
                </option>
                <option
                  value="1953"
                >
                  1953
                </option>
                <option
                  value="1954"
                >
                  1954
                </option>
                <option
                  value="1955"
                >
                  1955
                </option>
                <option
                  value="1956"
                >
                  1956
                </option>
                <option
                  value="1957"
                >
                  1957
                </option>
                <option
                  value="1958"
                >
                  1958
                </option>
                <option
                  value="1959"
                >
                  1959
                </option>
                <option
                  value="1960"
                >
                  1960
                </option>
                <option
                  value="1961"
                >
                  1961
                </option>
                <option
                  value="1962"
                >
                  1962
                </option>
                <option
                  value="1963"
                >
                  1963
                </option>
                <option
                  value="1964"
                >
                  1964
                </option>
                <option
                  value="1965"
                >
                  1965
                </option>
                <option
                  value="1966"
                >
                  1966
                </option>
                <option
                  value="1967"
                >
                  1967
                </option>
                <option
                  value="1968"
                >
                  1968
                </option>
                <option
                  value="1969"
                >
                  1969
                </option>
                <option
                  value="1970"
                >
                  1970
                </option>
                <option
                  value="1971"
                >
                  1971
                </option>
                <option
                  value="1972"
                >
                  1972
                </option>
                <option
                  value="1973"
                >
                  1973
                </option>
                <option
                  value="1974"
                >
                  1974
                </option>
                <option
                  value="1975"
                >
                  1975
                </option>
                <option
                  value="1976"
                >
                  1976
                </option>
                <option
                  value="1977"
                >
                  1977
                </option>
                <option
                  value="1978"
                >
                  1978
                </option>
                <option
                  value="1979"
                >
                  1979
                </option>
                <option
                  value="1980"
                >
                  1980
                </option>
                <option
                  value="1981"
                >
                  1981
                </option>
                <option
                  value="1982"
                >
                  1982
                </option>
                <option
                  value="1983"
                >
                  1983
                </option>
                <option
                  value="1984"
                >
                  1984
                </option>
                <option
                  value="1985"
                >
                  1985
                </option>
                <option
                  value="1986"
                >
                  1986
                </option>
                <option
                  value="1987"
                >
                  1987
                </option>
                <option
                  value="1988"
                >
                  1988
                </option>
                <option
                  value="1989"
                >
                  1989
                </option>
                <option
                  value="1990"
                >
                  1990
                </option>
                <option
                  value="1991"
                >
                  1991
                </option>
                <option
                  value="1992"
                >
                  1992
                </option>
                <option
                  value="1993"
                >
                  1993
                </option>
                <option
                  value="1994"
                >
                  1994
                </option>
                <option
                  value="1995"
                >
                  1995
                </option>
                <option
                  value="1996"
                >
                  1996
                </option>
                <option
                  value="1997"
                >
                  1997
                </option>
                <option
                  value="1998"
                >
                  1998
                </option>
                <option
                  value="1999"
                >
                  1999
                </option>
                <option
                  value="2000"
                >
                  2000
                </option>
                <option
                  value="2001"
                >
                  2001
                </option>
                <option
                  value="2002"
                >
                  2002
                </option>
                <option
                  value="2003"
                >
                  2003
                </option>
                <option
                  value="2004"
                >
                  2004
                </option>
                <option
                  value="2005"
                >
                  2005
                </option>
                <option
                  value="2006"
                >
                  2006
                </option>
                <option
                  value="2007"
                >
                  2007
                </option>
                <option
                  value="2008"
                >
                  2008
                </option>
                <option
                  value="2009"
                >
                  2009
                </option>
                <option
                  value="2010"
                >
                  2010
                </option>
                <option
                  value="2011"
                >
                  2011
                </option>
                <option
                  value="2012"
                >
                  2012
                </option>
                <option
                  value="2013"
                >
                  2013
                </option>
                <option
                  value="2014"
                >
                  2014
                </option>
                <option
                  value="2015"
                >
                  2015
                </option>
                <option
                  value="2016"
                >
                  2016
                </option>
                <option
                  value="2017"
                >
                  2017
                </option>
                <option
                  value="2018"
                >
                  2018
                </option>
                <option
                  value="2019"
                >
                  2019
                </option>
                <option
                  value="2020"
                >
                  2020
                </option>
              </select>
              <span
                class="icon select--chevron"
              >
                <i
                  class="icon -stroke -dark"
                >
                  <svg
                    height="1em"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <title>
                      ionicons-v5-a
                    </title>
                    <polyline
                      points="112 184 256 328 400 184"
                      style="fill: none; stroke-linecap: round; stroke-linejoin: round; stroke-width: 48;"
                    />
                  </svg>
                </i>
              </span>
            </div>
          </div>
        </div>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        Country of Birth
      </legend>
      <div
        class="formgroup--content"
      >
        <select
          class="select--native"
          disabled=""
          name="countryOfBirth"
        >
          <option>
            No options
          </option>
        </select>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        Nationality
      </legend>
      <div
        class="formgroup--content"
      >
        <select
          class="select--native"
          disabled=""
          name="nationality"
        >
          <option>
            No options
          </option>
        </select>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        Marital Status
      </legend>
      <div
        class="formgroup--content"
      >
        <select
          class="select--native"
          disabled=""
          name="maritalStatus"
        >
          <option>
            No options
          </option>
        </select>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        No. of Dependants
      </legend>
      <div
        class="formgroup--content"
      >
        <select
          class="select--native"
          disabled=""
          name="dependants"
        >
          <option>
            No options
          </option>
        </select>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        No. of Adults in Household
      </legend>
      <div
        class="formgroup--content"
      >
        <select
          class="select--native"
          disabled=""
          name="adultsInHousehold"
        >
          <option>
            No options
          </option>
        </select>
      </div>
    </fieldset>
    <fieldset
      class="formgroup"
    >
      <legend
        class="formgroup--legend"
      >
        Please Confirm
      </legend>
      <div
        class="formgroup--content"
      >
        <div
          class="checkbox--wrapper"
        >
          <input
            class="checkbox"
            id="consent"
            name="consent"
            type="checkbox"
            value=""
          />
          <label
            class="checkbox--label"
            for="consent"
          >
            <span
              class="text -regular -darker"
            >
              I wish to receive emails and SMS messages for updates on the latest deals, offers and promotions.
            </span>
          </label>
        </div>
        <div
          class="checkbox--wrapper"
        >
          <input
            class="checkbox"
            id="termsAndCons"
            name="termsAndCons"
            type="checkbox"
            value=""
          />
          <label
            class="checkbox--label"
            for="termsAndCons"
          >
            <span
              class="text -regular -darker"
            >
              agree to the terms and conditions.
            </span>
          </label>
        </div>
      </div>
    </fieldset>
    <button
      class="button -primary -regular -solid"
      tabindex="0"
      type="submit"
    >
      <div
        class="button--inner"
      >
        Continue
        <i
          class="icon -medium -small -stroke"
        >
          <svg
            height="1em"
            viewBox="0 0 512 512"
            width="1em"
          >
            <title>
              ionicons-v5-b
            </title>
            <path
              d="M256,48C141.13,48,48,141.13,48,256s93.13,208,208,208,208-93.13,208-208S370.87,48,256,48ZM216,374.63,193.37,352l96-96-96-96L216,137.37,334.63,256Z"
            />
          </svg>
        </i>
      </div>
    </button>
  </form>
</div>
`;
